Alire.Utils.User_Input: add function to query user's name, email and GitHub login

Fabien-Chouteau opened the pull request:
Can be used to implement #455.

Fabien-Chouteau committed:
Alire.Utils.User_Input: add function to query user's name, email and github login

mosteo commented on src/alire/alire-utils-user_input.adb:
```diff
@@ -201,4 +264,69 @@ package body Alire.Utils.User_Input is
+   function User_GitHub_Login return String
+   is (Config_Or_Query_String (Config_Key => "user.github_login",
+                               Question   => "Your GitHub login",
+                               Default    => "githib-username",
```
typo

mosteo replied:
(in "githib-username")

Fabien-Chouteau commented:
Elaboration issue :(

Sorry I don't have time to investigate before the week-end.

mosteo committed:
Break circular elaboration issue

Since Config uses Utils.User_Input indirectly via Deplyers.System, a
possibility is to move new User_Input functions that need Config into a new
package.

mosteo commented:
Hi Fabien, I took a stab at fixing the circular elab, I hope you don't mind.

Fabien-Chouteau commented:
Thank you :+1:
